### ISSUE TITLE
window union sql return invalid when call http api service

### ISSUE THREAD

**Issue by autobugfree** [4paradigm/OpenMLDB#3409]
**Bug Description**
In the official tutorial [page ](https://openmldb.ai/docs/zh/main/tutorial/tutorial_sql_2.html) , follow the tutorial to verify the following sql

```
  SELECT 
  id as out3id,
  sum(purchase_amt) over w7d_merchant as w7d_merchant_purchase_amt_sum,
  count(purchase_amt) over w7d_merchant as w7d_merchant_purchase_count,
  FROM (select id, mid, cardno as card, trans_time as purchase_time, 0.0 as purchase_amt, "" as purchase_type from t1) as t11
   window w7d_merchant as (UNION (select 0L as id, mid, card, purchase_time, purchase_amt, purchase_type from t2) as t22 PARTITION BY mid ORDER BY purchase_time ROWS_RANGE BETWEEN 30d PRECEDING AND 1 PRECEDING INSTANCE_NOT_IN_WINDOW)
```

The results of sql execution are as follows:

```
 -------- ------------------------------- -----------------------------
  out3id   w7d_merchant_purchase_amt_sum   w7d_merchant_purchase_count
 -------- ------------------------------- -----------------------------
  10       NULL                            0
  2        NULL                            0
  4        10.000000                       1
  9        105.000000                      3
  3        NULL                            0
  7        NULL                            0
  8        NULL                            0
  1        NULL                            0
  5        80.000000                       1
  6        142.000000                      5
 -------- ------------------------------- -----------------------------
```

then  deploy the sql as api service(Reproduce step 3), and curl api (Reproduce step 4), return invalid data, feature is null.
ps: request body data is table t1 id=6 raw data 
```
{"code":0,"msg":"ok","data":{"data":[[6,null,0]]}}
```

**Expected Behavior**

expect return

```
{"code":0,"msg":"ok","data":{"data":[[6,142.000000,5]]}}
```

**Relation Case**

after call api service return invalid result, execute above sql again, all feature is null.

<img width="1266" alt="image" src="https://github.com/4paradigm/OpenMLDB/assets/133122159/8c916fd3-9f15-4574-bbc9-f232444b6ecb">

above all data feature is null not 100% appear immediately, sometimes wait a few minutes, but all in the end it's like this.

**Steps to Reproduce**

0. download table [raw file ](https://openmldb.ai/download/tutorial_sql/tutoral_sql_data.zip) and start openmldb ;
```
docker run -it 4pdosc/openmldb:0.8.2 bash
/work/init.sh
/work/openmldb/bin/openmldb --zk_cluster=127.0.0.1:2181 --zk_root_path=/openmldb --role=sql_client
```
1.    create database & table;
```
create database tutorial;
use tutorial;
create table t1 (id bigint, uid string, mid string, cardno string, trans_time timestamp, trans_amt double, trans_type string, province string ,city string, label bool);
create table t2 (mid string, card string, purchase_time timestamp, purchase_amt double, purchase_type string);
create table s1 (id int, col1 string, std_ts timestamp);
create table s2 (id int, col1 string, std_ts timestamp);
```
2.  load data;
```
SET @@execute_mode='online';
load data infile 'file:///work/tutorial/t1.csv' into table tutorial.t1 options(format='csv', header=true, mode='append');
load data infile 'file:///work/tutorial/t2.csv' into table tutorial.t2 options(format='csv', header=true, mode='append');
load data infile 'file:///work/tutorial/s1.csv' into table tutorial.s1 options(format='csv', header=true, mode='append');
load data infile 'file:///work/tutorial/s2.csv' into table tutorial.s2 options(format='csv', header=true, mode='append');
```

3. deploy demo_service33;
```
deploy demo_service33  
 SELECT 
  id as out3id,
  sum(purchase_amt) over w7d_merchant as w7d_merchant_purchase_amt_sum,
  count(purchase_amt) over w7d_merchant as w7d_merchant_purchase_count,
  FROM (select id, mid, cardno as card, trans_time as purchase_time, 0.0 as purchase_amt, "" as purchase_type from t1) as t11
   window w7d_merchant as (UNION (select 0L as id, mid, card, purchase_time, purchase_amt, purchase_type from t2) as t22 PARTITION BY mid ORDER BY purchase_time ROWS_RANGE BETWEEN 30d PRECEDING AND 1 PRECEDING INSTANCE_NOT_IN_WINDOW)
```
 
4. call api service **demo_service33**;
```
curl http://127.0.0.1:9080/dbs/tutorial/deployments/demo_service33 -X POST -d'{"input": [[6, "UserA", "M1", "1111111", 1644210000000, 20.0, "CASH", "BJ", "BeiJing", true]]}'
```
 
**Device info**

CPU: Intel Core i7
OS: macos  13.2.1

**Comment by aceforeverd** [comment]
@autobugfree Hi, can you verify the result of 
```sql
show table status
```

**Comment by vagetablechicken** [comment]
I can reproduce it. And `desc <table>` and `SHOW JOBS FROM NAMESERVER` show that, t1 won't add index, t2 add index and load data, op is succeed. Case is not good?

**Comment by autobugfree** [comment]
> @autobugfree Hi, can you verify the result of
> 
> ```sql
> show table status
> ```

@aceforeverd  result of the sql :
<img width="1434" alt="image" src="https://github.com/4paradigm/OpenMLDB/assets/133122159/e1a1dae0-2fff-48ad-8a37-a17542b7c023">

**Comment by vagetablechicken** [comment]
@aceforeverd data is out-dated.

We can set index when create table, and insert to check ttl:
```
create database tutorial;
use tutorial;
Create table t2(mid string, card string, purchase_time timestamp, purchase_amt double, purchase_type string, index(key=mid,ts=purchase_time,ttl_type=absolute,ttl=30d));
SET @@execute_mode='online';
insert into t2 values("","",1691394294879,0.0,"");
insert into t2 values("","",1644303600000,1.0,""); -- 2022-02-08 15:00:00 too old
select * from t2;
```

**Comment by vagetablechicken** [comment]
@autobugfree could you set the ts value bigger in source csv files? e.g. 1723016681000, means 2024-08-07 15:44:41, won't be deleted by ttl(data ts should >= current_time-30d, otherwise delete). I'll update the source data later.

**Comment by autobugfree** [comment]
> create database tutorial;
> use tutorial;
> Create table t2(mid string, card string, purchase_time timestamp, purchase_amt double, purchase_type string, index(key=mid,ts=purchase_time,ttl_type=absolute,ttl=30d));
> SET @@execute_mode='online';
> insert into t2 values("","",1691394294879,0.0,"");
> insert into t2 values("","",1644303600000,1.0,""); -- 2022-02-08 15:00:00 too old
> select * from t2;

@vagetablechicken  Why is the result correct when the data is first imported to online? 
Is it because ttl has not been executed before deploy sql?

**Comment by autobugfree** [comment]
Is the default ttl of the online model database 30 days?

**Comment by aceforeverd** [comment]
tracking solution in #3428

**Comment by aceforeverd** [comment]
> Is the default ttl of the online model database 30 days?

@autobugfree  no, there is no default ttl for table t2, but deploy stmt created.

A DEPLOY SQL statement automatically create a TTL expire rule for table if not exists, based on what kind of SQL it is going to deployed. Here, the SQL has the window definition `ROWS_RANGE 30d preceding AND 1 preceding INSTANCE_NOT_IN_WINDOW`, then a TTL rule of 30d will created for table t2.

**Comment by autobugfree** [comment]
> > Is the default ttl of the online model database 30 days?
> 
> @autobugfree no, there is no default ttl for table t2, but deploy stmt created.
> 
> A DEPLOY SQL statement automatically create a TTL expire rule for table if not exists, based on what kind of SQL it is going to deployed. Here, the SQL has the window definition `ROWS_RANGE 30d preceding AND 1 preceding INSTANCE_NOT_IN_WINDOW`, then a TTL rule of 30d will created for table t2.

OK，got it !

**Comment by aceforeverd** [comment]
we may close the issue after @vagetablechicken update the testing data

**Comment by vagetablechicken** [comment]
> we may close the issue after @vagetablechicken update the testing data

Found that data and the graph in doc are matched, it's not a good idea to change the data. Current way is set old system time

**Comment by vagetablechicken** [comment]
No deploy sql in tutorial doc and source files, so no need to change doc and zip. After 0.8.3, we can set bias to inf when deploy, won't expire data.

**Comment by vagetablechicken** [comment]
closes by feature #3456
